src/UnitTests/ImageLoaders/IntelHex32/IntelHex32LoaderTests.cs

```diff
@@ -0,0 +1,176 @@
+﻿#region License
+/* 
+ * Copyright (C) 2017-2018 Christian Hostelet.
+ * inspired by work of:
+ * Copyright (C) 1999-2017 John Källén.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+
+using NUnit.Framework;
+using Reko.Core;
+using Reko.Core.Services;
+using Reko.ImageLoaders.IntelHex32;
+using Reko.UnitTests.Mocks;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Linq;
+using System.Text;
+
+namespace Reko.UnitTests.ImageLoaders.IntelHex32
+{
+    [TestFixture]
+    public class IntelHex32LoaderTests
+    {
+        private ServiceContainer sc;
+
+        [SetUp]
+        public void Setup()
+        {
+            sc = new ServiceContainer();
+            sc.AddService<DecompilerEventListener>(new FakeDecompilerEventListener());
+        }
+
+        [Test]
+        public void IntelHex32_ParserTests()
+        {
```

Comment:
This test is doing to much. It's testing several records for different things. Consider splitting it up into separate methdos, explaining exactly what is being tested in each case. E.g. `IntelHexParser_ThrowOnEmptyLine`, `IntelHexParser_ThrowOnInvalidHexChars` etc.

Reply:
Tests amended